### PR TITLE
Cloud Event fixes & Resource cleanup

### PR DESCRIPTION
**Pull request by cescoffier**
Fix typos in Cloud Events attributes and date-time parsing.

Also clean up some resources in the connector (a timer, DLQ producer) and make sure tests don't keep producers/consumers open after the test execution.

### CODE REVIEW

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaFailureHandler.java [hunk 1]
```diff
@@ -29,4 +29,8 @@ public static Strategy from(String s) {
 
     <K, V> CompletionStage<Void> handle(IncomingKafkaRecord<K, V> record, Throwable reason);
 
+    default void terminate() {
```
Comment (kenfinnigan) [inline smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaFailureHandler.java:32]: As the default is do nothing is this needed?

Or was it mostly to make explicit that this case doesn't require anything?

**Review by kenfinnigan — APPROVED** [review]
Retriggered CI as it seemed stuck

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaFailureHandler.java (hunk 1 above)
Comment (cescoffier) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/fault/KafkaFailureHandler.java:32]: The DLQ implementation overrides it, and close the producer.
Other strategies don't need to do anything.

**Comment by cescoffier** [comment]
@kenfinnigan Oupsy... forgot to update the documentation files. Should be better now.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**